### ISSUE TITLE
Liquids not animating with Optifine, Liquids not animating in tanks

### ISSUE THREAD

**Issue by vvnnif** [DimensionalDevelopment/VanillaFix#39]
The fire which shows when entities are burning is not animated unless there is a fire block persent nearby.

-----

Many animated items from Avaritia (latest version) aren't animated at all when VanillaFix is installed, items include singularities, neutronium ingots/nuggets, infinite ingots, sword of the cosmos and the endest pearl. (Note that only the non-hardcoded animations seem to be broken by VanillaFix)

-----

Non-flowing liquids aren't animated on the side at all when VanillaFix is installed.

**Comment by Runemoro** [comment]
Please make a single issue for issues that are related, such as animations being broken.

**Comment by Runemoro** [comment]
The entity on fire animation has already been fixed in the latest dev build (https://github.com/DimensionalDevelopment/VanillaFix/releases/latest).

I am not able to reproduce liquid animations being broken. Are you talking about vanilla water or modded liquid animations?

**Comment by vvnnif** [comment]
It seems like only vanilla liquids are affected, so lava and water. Also, my apologies for the title, this was actually my first time opening an issue on github :p

**Comment by TheSnowyChickens** [comment]
Hello,

Tinkers' Construct liquids aren't animated too.

![2018-05-27_23 18 24](https://user-images.githubusercontent.com/29357962/40590742-4f56bb58-6204-11e8-89e8-f6f12801d469.png)

**Comment by TheSnowyChickens** [comment]
![grafik](https://user-images.githubusercontent.com/29357962/40605694-7d0fdfde-6263-11e8-98fc-ddb19f09aca2.png)
My mod List (Yes, opti is installed)

**Comment by vvnnif** [comment]
After some further testing the liquid bug seems to weirder than I thought. It's quite hard to explain but I've made a video showcasing the issue. https://www.youtube.com/watch?v=XeD64tv-9nE&feature=youtu.be (Optifine isn't installed)

**Comment by Runemoro** [comment]
@vvnnif I'm not really sure what the bug is... I see both the sides and the top of all the liquids animating. The avaritia items also animate properly in your video...

**Comment by ShadowofCubes** [comment]
@Runemoro The singularities aren't animated and the liquids are only animated at specific viewing angles https://youtu.be/XeD64tv-9nE?t=20

**Comment by Runemoro** [comment]
@ShadowofCubes It's impossible for VanillaFix to break animations from certain angles only. Also, I don't see the texture not animating from certain angles. I actually can't really see anything in that video since they keep moving so much.

What do you mean by the singularities?

**Comment by ZombieHDGaming** [comment]
Figured out the Tinkers issue. The texture isn't animated in the smeltery for some reason, but it is animated in the world. Since that isn't such a big issue, I think it's safe to cross that off the list.

**Comment by TheSnowyChickens** [comment]
1 more example: Lava in OpenBlocks Tanks

If I focus the tank -> Lava is animated
If not -> not animated

**Comment by Runemoro** [comment]
@TheSnowyChickens Support for OpenBlocks tanks has been added in the latest dev build.

**Comment by Runemoro** [comment]
Support for Tinker's has also been added!

**Comment by Runemoro** [comment]
@vvnnif Vanilla liquids also animate correctly with latest Optifine and VanillaFix for me. Please comment on this issue if it's still a problem for you.

**Comment by Nordblum** [comment]
Just resolved this problem, wish I didn't have to spend few hours to do so though. 
Running Forge 1.12.2-14.23.5.2768
Using The Unholy Quadfecta of Optimization: Optifine, BetterFPS, FoamFix, VanillaFix.
Animated textures just didn't work. Tried disabling OptiFine - didn't help. Tried additionally disabling FoamFix - didn't help. Re-enabled them, changed B:textureFixes to false - got them working. All of the four mods are up-to-date. 
Please, help me help you find the cause. Maybe fixes from different mods are trying to do about the same thing and thus failing, maybe omething completely different. I will provide all the logs, dumps and other files you'll ask for, as long as the next guy who installs all 4 of these mods won't get to spend hours trying to get the textures to animate.

**Comment by ZombieHDGaming** [comment]
@Nordblum what exactly is your issue? Because I've done heavy testing and all my animated textures and fluids render properly without an issue.

**Comment by Nordblum** [comment]
@ZombieHDGaming I installed all 4 mods. Got into a fresh creative world. Didn't change a single setting, didn't touch a single line in mod configs - textures weren't animated. Water, Lava, Fire, Nether Portal - none of them. the only animated texture I could find is when I set myself on fire - that fire worked. I then started diffling with settings and combinations of them. Read of reddit that mipmapping isn't very cool with animated textures - checked twice and made sure it was disabled. Tried running all of these optimization mods, except FoamFix - still didn't work. Maybe it's some other mod breaking the animations, but for some reason, going into Mod Settings and setting textureFixes in VanillaFix' settings works. I turned it to false, restarted the game - texture animations, both vanilla and modded, work now.
We can go discuss this stuff on Discord if you'd like, that way when you show me that I'm completely wrong and the solution was simple all along I won't be made a fool of for everyone to see lmao.
